File: C++/0454-4sum-ii.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        int res = 0;
        unordered_map <int, int> hashmap;
        for(int i = 0; i < nums3.size(); i++){
            for(int j = 0; j < nums4.size(); j++){
                int sum = nums3[i] + nums4[j];
                hashmap[sum]++;
            }
        }

        for(int i = 0; i < nums1.size(); i++){
            for(int j = 0; j < nums2.size(); j++){
                auto tmp = hashmap.find(0 - nums1[i] - nums2[j]);
                if(tmp != hashmap.end()){
                    res += tmp->second;
                }
            }
        }
        return res;      
    }
};

File: Python/0454-4sum-ii.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        res = 0
        hash = {}
        for i in nums3:
            for j in nums4:
                sum = i + j
                if sum in hash:
                    hash[sum] += 1
                else:
                    hash[sum] = 1
        for i in nums1:
            for j in nums2:
                tmp = 0 - i - j
                if tmp in hash:
                    res += hash[tmp]
        return res